server_test.go

```diff
@@ -743,6 +744,42 @@ func TestServer_SetDefaultRetentionPolicy_ErrRetentionPolicyNotFound(t *testing.
 	}
 }
 
+// Ensure the server pre-creates shard groups as needed.
+func TestServer_PreCreateRetentionPolices(t *testing.T) {
+	c := NewMessagingClient()
+	s := OpenServer(c)
+	defer s.Close()
+	s.CreateDatabase("foo")
+	s.CreateRetentionPolicy("foo", &influxdb.RetentionPolicy{Name: "mypolicy", Duration: 60 * time.Minute})
+
+	// Create two shard groups for the the new retention policy -- 1 which will age out immediately
+	// the other in more than an hour.
+	s.CreateShardGroupIfNotExists("foo", "mypolicy", time.Now().Add(-2*time.Hour))
+
+	// Check the two shard groups exist.
+	var g []*influxdb.ShardGroup
+	g, err := s.ShardGroups("foo")
+	if err != nil {
+		t.Fatal(err)
+	} else if len(g) != 1 {
+		t.Fatalf("expected 1 shard group but found %d", len(g))
+	}
+
+	// Run shard group pre-create.
+	s.ShardGroupPreCreate(time.Hour)
+
+	// Ensure enforcement is in effect across restarts.
+	s.Restart()
+
+	// First shard group should have been removed.
+	g, err = s.ShardGroups("foo")
```

Comment:
This comment doesn't sound like pre-creation is being tested. To simplify this test, I would remove all references to shard groups being expired. Have this test focus on pre-creation, forget about groups aging out.